external/mpicpp-lite/impl/Communicator.h

```diff
@@ -264,6 +264,24 @@ class Communicator {
     template <typename T>
     void gather(const T * in_values, int n, std::vector<T> & out_values, int root) const;
 
+    /// Gather together values from a group of processes
+    ///
+    /// @tparam T C++ datatype
+    /// @param in_values[in] Values to send
+    /// @param out_values[out] Buffer to receive the data
+    /// @param out_counts[in] Integer array (of length group size) containing the number of elements
+    ///        that are to be received from each process
+    /// @param out_offsets[in] Integer array (of length group size). Entry `i` specifies the
+    ///        displacement (relative to out_values) at which to place the incoming data from
+    ///        process `i`
+    /// @param root Rank of receiving process
+    template <typename T>
+    void gather(const std::vector<T> & in_values,
+                std::vector<T> & out_values,
+                const std::vector<int> & out_counts,
+                const std::vector<int> & out_offsets,
+                int root) const;
+
```

Comment:
If you are updating `mpicpp_lite`, please send a PR into [upstream](https://github.com/andrsd/mpicpp-lite) first.

Reply:
Will do. In other codes I work in, the workflow is to patch locally, push upstream, and then pull the new upstream version when it's available. This way we don't block development. Guess that really doesn't apply in this case. 😆

Reply:
If you are at the mercy of the package maintainer, that makes sense. They may be slow in merging and possibly releasing new versions. But, it can create conflicts because your local copy can diverge from the upstream, so next time you are updating the local copy, you may need to resolve such conflicts. Updating upstream first would prevent that... That's what I was thinking. But, we can keep this in the PR, there was no development in `mpicpp-lite` lately, so there should be no conflicts...

Note, that for your PR into upstream, I would ask you for a little test case, so we know this works.

Reply:
I'll submit a PR with a test case.